models/Element/Tag.php

```diff
@@ -319,4 +342,51 @@ public function correctPath()
             $this->idPath = '/';
         }
     }
+
+    /**
+     * @return mixed
+     */
+    public function getTranslations() {
+        return $this->translations ? unserialize($this->translations) : [];
+    }
+
+    /**
+     * @param mixed $translations
+     */
+    public function setTranslations($translations) {
+        $this->translations = $translations;
+    }
+
+    /**
+     * @return int
+     */
+    public function getCreationDate() {
+        return $this->creationDate;
+    }
+
+    /**
+     * @param int $creationDate
+     */
+    public function setCreationDate($creationDate) {
+        $this->creationDate = $creationDate;
+    }
+
+    /**
+     * @return int
+     */
+    public function getModificationDate() {
+        return $this->modificationDate;
+    }
+
+    /**
+     * @param int $modificationDate
+     */
+    public function setModificationDate($modificationDate) {
+        $this->modificationDate = $modificationDate;
+    }
+
+    public function __toString(){
+        $currentLanguage = \Pimcore::getContainer()->get('pimcore.locale')->getLocale();
+        return $this->getTranslations()['_' . $currentLanguage] ?? $this->name;
```

Comment:
This won't work if there are no translations at all, because in this case `$this->getTranslations()` returns an empty array while `?? $this->name` only gets called for `null`.

```suggestion
        return $this->getTranslations()['_' . $currentLanguage] ?: $this->name;
```

Reply:
This will lead to an "undefined index" error, when requesting non-existent translations whereas ?? does not

Reply:
Oh, you're totally right. `??` is correct here!